src/protocol/context/malicious.rs

```diff
@@ -77,9 +77,12 @@ impl<'a, F: Field> MaliciousContext<'a, F> {
     pub async fn upgrade_bit_triple(
         &self,
         record_id: RecordId,
+        bit_index: u32,
```

Comment:
Added `bit_index`. Calling `upgrade_bit_triple()` multiple times per bit panics, because it uses the same `upgrade_ctx`. This needs to be narrowed per bit like `upgrade_bit()`.